docs: added support for commit-copy-action (#1241)
harness/canary: apps/portal/src/content/docs/components/commit-copy-action.mdx

Reviewer: The component is `CommitCopyActions` not `CommitCopyAction`

Reviewer: Can we rename this file to match the name of the component?

---

> title: Commit Copy Action

Reviewer: ```suggestion
title: CommitCopyActions
```

> description: Commit Copy Action component documentation

Reviewer: ```suggestion
description: CommitCopyActions component
```

---

> The Commit Copy Action component provides a UI element for displaying a commit SHA and copying it to the clipboard. It also allows navigation to the commit details page.

Reviewer: ```suggestion
The `CommitCopyActions` component provides a UI element for displaying a commit SHA and copying it to the clipboard. It also allows navigation to the commit details page.
```

import { DocsPage } from "@/components/docs-page";

<DocsPage.ComponentExample
  client:only

>   code={`
>   <div className="flex">
>     <CommitCopyActions sha="abc1234" toCommitDetails={({ sha }) => "/commit/" + sha} />
>   </div>`}

Reviewer: ```suggestion
  code={`<div className="flex">
    <CommitCopyActions sha="abc1234" toCommitDetails={({ sha }) => "/commit/" + sha} />
</div>`}
```

Reviewer: Not something to fix in this PR, but the layout of the component should be baked into the component. Must revisit this at some point and bring the flex layout into the component itself.

/>

## Usage

```typescript jsx
import { CommitCopyActions } from '@harnessio/ui/components'

> 

Reviewer: Remember the ellipsis to make it obvious that this is not a code-complete example

```suggestion

// ...

```

return (
  <div className="flex">
    <CommitCopyActions
      sha="abc1234"
      toCommitDetails={({ sha }) => `/commit/${sha}`}

>       className="my-class"

Reviewer: This component doesn't accept a `className` component.
```suggestion
```

    />
  </div>
)
```

> ## API Reference

Reviewer: Can we add a short paragraph about the usage of the component and an example with the prop comments?

<DocsPage.PropsTable
  props={[
    {
      name: "sha",
      description:
        "The commit SHA to display and copy. This will be shown in the UI and can be copied to clipboard.",
      required: true,
      value: "string",
    },
    {
      name: "toCommitDetails",
      description:
        "Function that takes the SHA and returns a URL for navigating to the commit details page",
      required: false,
      value: "({ sha }: { sha: string }) => string",
    },

>     {
>       name: "className",
>       description: "Additional CSS classes to apply to the component",
>       required: false,
>       value: "string",
>     },

Reviewer: This prop doesn't exist
```suggestion
```

---
title: CommitCopyActions
description: CommitCopyActions component
---

The `CommitCopyActions` component provides a UI element for displaying a commit SHA and copying it to the clipboard. It also allows navigation to the commit details page.

import { DocsPage } from "@/components/docs-page";

<DocsPage.ComponentExample
  client:only
  code={`<div className="flex">
    <CommitCopyActions sha="abc1234" toCommitDetails={({ sha }) => "/commit/" + sha} />
</div>`}
/>

## Usage

```typescript jsx
import { CommitCopyActions } from '@harnessio/ui/components'

// ...

return (
  <div className="flex">
    <CommitCopyActions
      sha="abc1234"
      toCommitDetails={({ sha }) => `/commit/${sha}`}
    />
  </div>
)
```

## API Reference

The `CommitCopyActions` component provides action to copy commit sha to clipboard and navigate to commit details page.

```typescript jsx
<div className="flex">
  <CommitCopyActions

>     sha="abc1234"
>     toCommitDetails={({ sha }) => `/commit/${sha}`}

Reviewer: Can we add the inline comments to describe the props?
```suggestion
    sha="abc1234"                                   // SHA hash to copy to clipboard
    toCommitDetails={({ sha }) => `/commit/${sha}`} // callback to get the unique URL to navigate to on click
```
